Makefile

```diff
@@ -262,15 +262,15 @@ shellcheck: $(SHELLCHECK)
 ## --------------------------------------
 .PHONY: build
 build:
-	GOPROXY=$(GOPROXY) CGO_ENABLED=0 GOOS=linux go build -mod=vendor -a -ldflags $(LDFLAGS) -o _output/secrets-store-csi ./cmd/secrets-store-csi-driver
+	GOPROXY=$(GOPROXY) CGO_ENABLED=1 GOEXPERIMENT=strictfipsruntime GOOS=linux go build -mod=vendor -a -tags "strictfipsruntime,openssl" -ldflags $(LDFLAGS) -o _output/secrets-store-csi ./cmd/secrets-store-csi-driver
 
 .PHONY: build-e2e-provider
 build-e2e-provider:
-	cd test/e2eprovider && GOPROXY=$(GOPROXY) CGO_ENABLED=0 GOOS=linux go build -mod=vendor -a -tags "e2e" -o e2e-provider
+	cd test/e2eprovider && GOPROXY=$(GOPROXY) CGO_ENABLED=1 GOEXPERIMENT=strictfipsruntime GOOS=linux go build -mod=vendor -a -tags "strictfipsruntime,openssl,e2e" -o e2e-provider
 
 .PHONY: build-windows
 build-windows:
-	GOPROXY=$(GOPROXY) CGO_ENABLED=0 GOOS=windows go build -mod=vendor -a -ldflags $(LDFLAGS) -o _output/secrets-store-csi.exe ./cmd/secrets-store-csi-driver
+	GOPROXY=$(GOPROXY) CGO_ENABLED=0 GOOS=windows go build -a -ldflags $(LDFLAGS) -o _output/secrets-store-csi.exe ./cmd/secrets-store-csi-driver
```

Comment:
Since we are not building/releasing a Windows-based image, I think there is no need to have `-mod=vendor` as a carry change from upstream.